### PR TITLE
[#7][Chore] Configure UI tests

### PR DESCRIPTION
**Pull request by malparty**
- #7 

## What happened 👀

- [x] Setup cypress and its GitHub actio — following our compass conventions 
- [x] Add a first UI test to confirm it is working well

## Insight 📝

> Bigger than expected! (in terms of features) :) 

## Proof Of Work 📹

|Tests can fail | Test can pass|
|---|---|
|image|image|

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (rafayet-monon) [inline cypress/fixtures/example.json]: Do we need this file?

File: cypress/plugins/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="cypress" />
```
Comment (rafayet-monon) [inline cypress/plugins/index.js:1]: Should we get rid of the comments?

File: cypress/fixtures/example.json (hunk 1 above)
Comment (malparty) [reply cypress/fixtures/example.json]: ✅ Done in 7821c13 ;-)

File: cypress/plugins/index.js (hunk 2 above)
Comment (malparty) [reply cypress/plugins/index.js:1]: ✅ Done in 7821c13

PS: in 3c9876c I did add the data-test-id config ;-)
This line enables the typescript compiler (and IntelliSense) to get all the type signatures related to cypress (ex: doing cy.XXXX).

Comment (malparty) [reply cypress/plugins/index.js:1]: (note I kept the /// as this is not a comment but a type reference to enable typescript autocompletion with cypress type signatures).

File: cypress/integration/UserAuthentication/login.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  const LOGIN_SCREEN_CONTAINER_SELECTOR = '[data-test-id=login-screen-container]';
+
+  it('Successfully loads', () => {
+    cy.visit('/');
+  });
+
+  it('Displays a "Sign in to Nimble" message', () => {
+    cy.visit('/');
+
+    expect(cy.get(LOGIN_SCREEN_CONTAINER_SELECTOR).contains('Sign in to Nimble'));
```
Comment (carryall) [inline cypress/integration/UserAuthentication/login.spec.ts]: How about using `findByTestId` from [@testing-library/cypress](https://docs.cypress.io/faq/questions/using-cypress-faq#Can-I-use-Testing-Library) here?💡 

```js
const LOGIN_SCREEN_CONTAINER_SELECTOR = 'login-screen-container';

//...

expect(cy.findByTestId(LOGIN_SCREEN_CONTAINER_SELECTOR).contains('Sign in to Nimble'));
```

File: package.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "@nimblehq/eslint-config-nimble": "2.1.1",
     "@pmmmwh/react-refresh-webpack-plugin": "0.4.3",
     "@svgr/webpack": "5.5.0",
+    "@testing-library/cypress": "^8.0.1",
```
Comment (carryall) [inline package.json]:
```suggestion
    "@testing-library/cypress": "8.0.1",
```

Comment (malparty) [reply package.json]: ✅ Fixed in 78eb180 :)

File: cypress/integration/UserAuthentication/login.spec.ts (hunk 3 above)
Comment (malparty) [reply cypress/integration/UserAuthentication/login.spec.ts]: I had to add an import in the config, and a ref in the file -- to support type-check in VS Code. 

`import '@testing-library/cypress/add-commands';` (in cypress/support/commands.js)

But it's now working well :) 
Thanks!

✅ Done in 78eb180

**Review by carryall — APPROVED** [review]

**Review by rafayet-monon — APPROVED** [review]

**Pull request merged**